src/goto-programs/restrict_function_pointers.cpp

```diff
@@ -193,6 +198,51 @@ void handle_call(
                        address_of_exprt{function_pointer_target_symbol_expr}}));
   }
 }
+
+void get_by_name_restriction(
```

Comment:
:no_entry_sign: I think it is suprising that a function called `get_...` doesn't return anything, can this be structured in a way so that it actually does get something? Otherwise, a comment would help

Reply:
Yeah I think it'd be much clearer to return a new restrictions (and append at the callsite if required)

Reply:
The function now returns an `optionalt<restrictiont>`.